sound/soc/intel/boards/cml_rt1011_rt5682.c

```diff
@@ -547,8 +547,8 @@ static int snd_cml_rt1011_probe(struct platform_device *pdev)
 	if (sof_rt1011_quirk & (SOF_RT1011_SPEAKER_TL |
 				SOF_RT1011_SPEAKER_TR)) {
 		for_each_card_prelinks(&snd_soc_card_cml, i, dai_link) {
-			if (!strcmp(cml_rt1011_rt5682_dailink[i].codecs[0].dai_name,
-				    CML_RT1011_CODEC_DAI)) {
+			if (!strncmp(cml_rt1011_rt5682_dailink[i].codecs[0].dai_name,
+				     CML_RT1011_CODEC_DAI, sizeof(CML_RT1011_CODEC_DAI) - 1)) {
```

Comment:
not sure we need the -1?

Reply:
sizeof include null character also

Reply:
I know but I have never seen a strncmp() with sizeof -1...

Reply:
"rt1011-aif" sizeof=11 strlen=10

I grep strncmp in kernel tree, i found good number of reference. For example, 
https://github.com/thesofproject/linux/blob/topic/sof-dev/kernel/gcov/fs.c#L357

Reply:
In this case, I think what we need is ``sizeof(CML_RT1011_CODEC_DAI)``. We want to match the codec dai_name exactly the same as "rt1011-aif". ie. the 11th char of codec dai name must be null. strncmp("rt1011-aif1", "rt1011-aif", 10) will return 0 and that is not what we want.

Reply:
Ah, I missed that point, makes sense!